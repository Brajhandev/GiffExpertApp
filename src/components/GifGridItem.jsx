import PropTypes from "prop-types";

const GifGridItem = ({ title, url }) => {
  return (
    <div>
      <p>{title}</p>
      <img src={url} alt="" />
    </div>
  );
};

export default GifGridItem;

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
