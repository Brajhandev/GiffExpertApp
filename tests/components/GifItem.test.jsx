import { render, screen } from "@testing-library/react";
import GifGridItem from "../../src/components/GifGridItem";

describe("Pruebas en <GifItem/>", () => {
  const title = "Saitama";
  const url = "https://one-punch.com/saitama.jpg";

  test("debe hacer match con el snapshot", () => {
    const { container } = render(<GifGridItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe msotrar la imagen con el URL y el ALT indicado", () => {
    render(<GifGridItem title={title} url={url} />);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });
});
