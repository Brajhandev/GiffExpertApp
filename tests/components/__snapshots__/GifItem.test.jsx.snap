// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div>
    <p>
      Saitama
    </p>
    <img
      alt=""
      src="https://one-punch.com/saitama.jpg"
    />
  </div>
</div>
`;
